Build failure: victoriametrics

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix-build -A victoriametrics --argstr system aarch64-linux`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/313274963

### Relevant log output

```console
/nix/store/ybh1p0f6sd3i7b66chf3gd8b1yy41p0c-go-1.25.3/share/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
/nix/store/hwbs2xmw7jpmcdxs3kjcla0iprv28bj9-gcc-wrapper-14.3.0/bin/gcc -s -Wl,-z,relro -pie -o $WORK/b001/exe/a.out -Wl,--export-dynamic-symbol=_cgo_panic -Wl,--export-dynamic-symbol=_cgo_topofstack -Wl,--export-dynamic-symbol=crosscall2 -Wl,--compress-debug-sections=zlib /build/go-link-1986605131/go.o /build/go-link-1986605131/000000.o /build/go-link-1986605131/000001.o /build/go-link-1986605131/000002.o /build/go-link-1986605131/000003.o /build/go-link-1986605131/000004.o /build/go-link-1986605131/000005.o /build/go-link-1986605131/000006.o /build/go-link-1986605131/000007.o /build/go-link-1986605131/000008.o /build/go-link-1986605131/000009.o /build/go-link-1986605131/000010.o /build/go-link-1986605131/000011.o /build/go-link-1986605131/000012.o /build/go-link-1986605131/000013.o /build/go-link-1986605131/000014.o /build/go-link-1986605131/000015.o /build/go-link-1986605131/000016.o /build/go-link-1986605131/000017.o /build/go-link-1986605131/000018.o /build/go-link-1986605131/000019.o /build/go-link-1986605131/000020.o /build/go-link-1986605131/000021.o /build/go-link-1986605131/000022.o /build/go-link-1986605131/000023.o /build/go-link-1986605131/000024.o /build/go-link-1986605131/000025.o /build/go-link-1986605131/000026.o /build/go-link-1986605131/000027.o /build/go-link-1986605131/000028.o -O2 -g -lresolv -O2 -g -lpthread -O2 -g /build/source/vendor/github.com/valyala/gozstd/libzstd_linux_arm64.a -O2 -g
/nix/store/m4wy5f21055ninhpdq51mv7chjj2qk98-binutils-2.44/bin/ld: /build/source/vendor/github.com/valyala/gozstd/libzstd_linux_arm64.a(zdict.o): relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `stderr@@GLIBC_2.17' which may bind externally can not be used when making a shared object; recompile with -fPIC
/nix/store/m4wy5f21055ninhpdq51mv7chjj2qk98-binutils-2.44/bin/ld: /build/source/vendor/github.com/valyala/gozstd/libzstd_linux_arm64.a(zdict.o)(.text+0x274): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `stderr@@GLIBC_2.17'
/nix/store/m4wy5f21055ninhpdq51mv7chjj2qk98-binutils-2.44/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

### Additional context

Currently running a bisect. I already know it's part of the last staging-next (de21549a840be528ef4763dd7dca2006caf9c11f).

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.57, NixOS, 25.11 (Xantusia), 25.11.20251110.b549734`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/cb0iy8160z0s8dy06vmdp33pwdk1whh1-source`

### Notify maintainers

@yorickvP @ivan @leona-ya @Shawn8901 @ryan4yin

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/vi/victoriametrics/package.nix
+++ b/pkgs/by-name/vi/victoriametrics/package.nix
@@ -23,6 +23,7 @@ buildGoModule (finalAttrs: {
   };
 
   vendorHash = null;
+  env.CGO_ENABLED = 0;
 
   subPackages =
     lib.optionals withServer [

